docs/concepts/what-is-a-codec-server.md

```diff
@@ -0,0 +1,41 @@
+---
+id: what-is-a-codec-server
+title: What is a Codec Server?
+sidebar_label: Codec Server
+description: Explanation and implementation of a remote encryption/decryption server.
+---
+
+A Codec Server is a feature that can perform additional levels of encoding and decoding on Payloads that are handled by [tctl](/tctl/index) or the [Web UI](/web-ui/how-to-use-a-list-filter-in-the-temporal-web-ui).
+
+- [How to set up a Codec Server](/clusters/how-to-set-up-codec-server/)
+
+The Web UI and tctl both use a default [Data Converter](/concepts/what-is-a-data-converter), which is capable of serialization only.
+
+Codec Servers can encrypt, compress, and change the format of a Payload object.
+These measures can further secure your data.
+
+![](/img/tctl-diagram-codec-server.svg)
+
+## Use case: tctl
+
+Suppose that you want to view Workflow History.
+This information needs to be decoded before it can be viewed.
+
+You can use [tctl workflow showid](/tctl/workflow/showid) to view a Workflow Execution Event History.
+
+```bash
+    tctl workflow showid <workflowID>
+```
+
+With a Codec Server, the Payload is decoded after being deserialized by tctl's default Data Converter. The default Data Converter sends the Payload to a given endpoint, and receives a decoded Payload if the API returns a successful result.
```

Comment:
The Payload is decoded before being deserialised. For example if the Payload has been encrypted, it would need to be decrypted before the DataConverter can deserialise it.